src/components/tree-detail/tree-water-needs/contact-dialog.tsx

```diff
@@ -0,0 +1,216 @@
+/* eslint-disable max-lines */
+import React, { useCallback, useState } from "react";
+import { useI18nStore } from "../../../i18n/i18n-store";
+import { AlertDialog } from "../../alert-dialog/alert-dialog";
+import { PrimaryButton } from "../../buttons/primary";
+import { TertiaryButton } from "../../buttons/tertiary";
+import { CheckIcon } from "../../icons/check-icon";
+import { CloseIcon } from "../../icons/close-icon";
+import Markdown from "react-markdown";
+import { supabaseClient } from "../../../auth/supabase-client";
+
+const closeContactDialog = () => {
+	(document.getElementById("contact-dialog") as HTMLDialogElement).close();
+};
+
+interface ContactDialogProps {
+	contactUsername: string;
+}
+export const ContactDialog: React.FC<ContactDialogProps> = ({
+	contactUsername,
+}) => {
+	const [message, setMessage] = useState("");
```

Comment:
I don't think we need this message state. We can get the message from the form submit event, which would reduce renderings of this component while typing the message you want to send. Or am I missing something?

Reply:
In general you are right, but we discussed to also check for links in the text so we need this message state. We disable the "Send" button as long as we find some links in the message. (to be implemented)

Reply:
Makes sense 👍 then let's keep it for the moment.